engine/src/flutter/impeller/entity/contents/atlas_contents.h

```diff
@@ -64,6 +105,38 @@ class AtlasContents final : public Contents {
   AtlasContents& operator=(const AtlasContents&) = delete;
 };
 
+/////
+
+class ColorFilterAtlasContents final : public Contents {
+ public:
+  explicit ColorFilterAtlasContents();
+
+  ~ColorFilterAtlasContents() override;
+
+  void SetGeometry(AtlasGeometry* geometry);
+
+  void SetAlpha(Scalar alpha);
+
+  void SetMatrix(ColorMatrix matrix);
+
+  // |Contents|
+  std::optional<Rect> GetCoverage(const Entity& entity) const override;
+
+  // |Contents|
+  bool Render(const ContentContext& renderer,
+              const Entity& entity,
+              RenderPass& pass) const override;
+
+ private:
+  AtlasGeometry* geometry_ = nullptr;
```

Comment:
Add comment why raw pointer can't result in a dangling pointer.

Reply:
Done